test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -659,12 +662,55 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
 
                     // Wait for error message to be logged.
                     await logTcs.Task.TimeoutAfter(TimeSpan.FromSeconds(10));
+
+                    // The server should close the connection in this situation.
+                    await connection.WaitForConnectionClose().TimeoutAfter(TimeSpan.FromSeconds(10));
                 }
             }
 
-            Assert.Equal(
-                $"Response Content-Length mismatch: too few bytes written (12 of 13).",
-                errorMessage);
+            mockTrace.Verify(trace =>
+                trace.ApplicationError(
+                    It.IsAny<string>(),
+                    It.Is<InvalidOperationException>(ex =>
+                        ex.Message.Equals($"Response Content-Length mismatch: too few bytes written (12 of 13).", StringComparison.Ordinal))));
+        }
+
+        [Fact]
+        public async Task WhenAppWritesLessThanContentLengthButRequestIsAbortedErrorNotLogged()
+        {
+            var abortTcs = new TaskCompletionSource<object>();
+            var mockTrace = new Mock<IKestrelTrace>();
+
+            using (var server = new TestServer(async httpContext =>
+            {
+                httpContext.RequestAborted.Register(() =>
+                {
+                    abortTcs.SetResult(null);
+                });
+                httpContext.Response.ContentLength = 12;
+                await httpContext.Response.WriteAsync("hello,");
+
+                // Wait until request is aborted so we know Frame will skip the response content length check.
+                await abortTcs.Task.TimeoutAfter(TimeSpan.FromSeconds(10));
```

Comment:
Would this test still pass of this timed out after 10s without being aborted? If so, make it so it fails if the request doesn't get aborted. I hat tests that time out and silently pass.